src/EditorFeatures/VisualBasic/LineCommit/CommitBufferManager.vb

```diff
@@ -114,6 +114,23 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit
                         Return
                     End If
 
+                    If Not isExplicitFormat Then
+                        ' Avoid automatically formatting extremely large dirty spans without an explicit request from
+                        ' the user. The "large span threshold" is 7000 lines. The 7000 line threshold is an estimated
+                        ' value accounting for a lower-bound of the algorithmic complexity of text differencing in
+                        ' designer cases along with measurements of a pathological example demonstrated at 14000 lines.
+                        ' We expect Windows Forms designer formatting operations to run in under ~15 seconds on average
+                        ' current hardware when nearing the threshold.
```

Comment:
It also might be worth noting pre-Roslyn had a similar check.

Reply:
I will include it in the bar check for 15.9 but it doesn't seem relevant for future investigations beyond what's available in the bug reports.